api/build.gradle

```diff
@@ -167,7 +178,16 @@ dependencies {
 
 }
 
-test {
+task integration(type: Test) {
+  group = LifecycleBasePlugin.VERIFICATION_GROUP
+  testClassesDirs = sourceSets.integration.output.classesDirs
+  classpath = sourceSets.integration.runtimeClasspath
+  // Option to control size of stack trace:
+  // jvmArgs '-XX:MaxJavaStackTraceDepth=10'
+}
+
+tasks.withType(Test) {
+  outputs.upToDateWhen { false }
```

Comment:
When you request to have tests run, they should run, even if the code is up-to-date.

Reply:
I don't mind this so much but... is there a reason to re-run tests if the code hasn't changed?

Reply:
You're right. This only applies to tests that have or rely on side-effects. Made this a setting for integration instead.